src/mongoose_flatlog_formatter.erl

```diff
@@ -0,0 +1,13 @@
+-module(mongoose_flatlog_formatter).
+-export([format/2]).
+
+format(Map, UsrConfig = #{template_name := Name}) ->
```

Comment:
I only fear the performance regression that aaaall this tremendous nesting might cause. I'm paranoid about performance. We'll see in the future.